src/apps/service/apps/de/jobs/common.clj

```diff
@@ -52,45 +53,14 @@
        (mapcat (partial build-environment-entries config default-values))
        (into {})))
 
-(defn- filter-min-requirement-keys
-  [m]
-  (select-keys m [:min_memory_limit
-                  :min_cpu_cores
-                  :min_disk_space]))
-
-(defn- filter-container-max-requirements
-  [container]
-  (select-keys container [:memory_limit
-                          :max_cpu_cores]))
-
-(defn- filter-request-max-requirements
-  "Ensure max requirement requests from the client are at least 0,
-  so that any max set by a tool is not automatically requested,
-  then the job services can choose a reasonable default instead."
-  [requirements]
-  {:memory_limit  (get requirements :memory_limit 0)
-   :max_cpu_cores (get requirements :max_cpu_cores 0)})
-
-(defn- limit-container-min-requirement
-  [container req-key-min req-key-max]
-  (if (and (contains? container req-key-min)
-           (contains? container req-key-max))
-    (assoc container req-key-min (min (get container req-key-min)
-                                      (get container req-key-max)))
-    container))
-
 (defn- reconcile-container-requirements
   "reconcile submission requirement requests with tool requirements"
   [container requirements]
-  (-> (merge container
-             (merge-with max
-                         (filter-min-requirement-keys container)
-                         (filter-min-requirement-keys requirements))
-             (merge-with min
-                         (filter-container-max-requirements container)
-                         (filter-request-max-requirements requirements)))
-      (limit-container-min-requirement :min_memory_limit :memory_limit)
-      (limit-container-min-requirement :min_cpu_cores :max_cpu_cores)))
+  (assoc container
+         :min_memory_limit (resources/get-required-memory container requirements)
+         :memory_limit     (resources/get-max-memory container requirements)
+         :min_cpu_cores    (resources/get-required-cpus container requirements)
+         :max_cpu_cores    (resources/get-max-cpus container requirements)))
```

Comment:
This feels a lot less elegant than the code that was there before, but I wanted to make things explicit in order to make it easier to reason about the requests and limits.